content/docs/rfcs/26/README.md

```diff
@@ -0,0 +1,152 @@
+Specifically, it describes how encryption, decryption and signing works in [6/WAKU1](/spec/6) and in [10/WAKU2](/spec/10) with [14/WAKU-MESSAGE version 1](/spec/14/#version1).
+
+This specification effectively replaces [7/WAKU-DATA](/spec/7) as well as [6/WAKU1 Payload encryption](/spec/6/#payload-encryption) but written in a way that is agnostic and self-contained for Waku v1 and Waku v2.
+
+Large sections of the specification originate from [EIP-627: Whisper spec](https://eips.ethereum.org/EIPS/eip-627) as well from [RLPx Transport Protocol spec (ECIES encryption)](https://github.com/ethereum/devp2p/blob/master/rlpx.md#ecies-encryption) with some modifications.
+
+## Design requirements
+
+- *Confidentiality*: The adversary should not be able to learn what data is being sent from one Waku node to one or several other Waku nodes.
+- *Authenticity*: The adversary should not be able to cause Waku endpoint to accept data from any third party as though it came from the other endpoint.
+- *Integrity*: The adversary should not be able to cause a Waku endpoint to accept data that has been tampered with.
+
+Notable, *forward secrecy* is not provided for at this layer.
+If this property is desired, a more fully featured secure communication protocol can be used on top, such as [Status 5/SECURE-TRANSPORT](https://specs.status.im/spec/5).
+
+It also provides some form of *unlinkability* since:
+- only participants who are able to decrypt a message can see its signature
+- payload are padded to a fixed length
+
+## Cryptographic primitives
+
+- AES-256-GCM
+- ECIES
+- ECSDA
+- KECCAK-256
+
+ECIES is using the following cryptosystem:
+- Curve: secp256k1
+- KDF: NIST SP 800-56 Concatenation Key Derivation Function
+- MAC: HMAC with SHA-256
+- AES: AES-128-CTR
+
+## Specification
+
+For 6/WAKU1, the `data` field is used in the `waku envelope`, and the field MUST contain the encrypted payload.
+
+For 10/WAKU2, the `payload` field is used in `WakuMessage` and MUST contain the encrypted payload.
+
+The fields that are concatenated and encrypted as part of the field are:
+ - flags
+ - auxiliary field
+ - payload
+ - padding
+ - signature
+
+### ABNF
+
+Using [Augmented Backus-Naur form (ABNF)](https://tools.ietf.org/html/rfc5234) we have the following format:
+
+```abnf
+; 1 byte; first two bits contain the size of auxiliary field, 
+; third bit indicates whether the signature is present.
+flags           = 1OCTET
+
+; contains the size of payload.
+payload-length  = 4*OCTET
+
+; byte array of arbitrary size (may be zero).
+payload         = *OCTET
+
+; byte array of arbitrary size (may be zero).
+padding         = *OCTET
+
+; 65 bytes, if present.
+signature       = 65OCTET
+
+; 12 bytes, if present (in case of symmetric encryption).
+iv              = 12OCTET
+
+; 16 bytes, if present (in case of symmetric encryption).
+tag             = 16OCTET
+
+data            = flags auxiliary-field payload padding [signature]
```

Comment:
I believe this is changed now

Reply:
From before? correct yeah, it *could* be wrong but from what I can tell in the nim-eth / geth code it was largely confusing, because it made "salt" seem special, as opposed to integrated with AES-GCM appended tag+iv. Ditto for asymmetric case. Removing it seems to make the spec and behavior cleaner, assuming it is accurate.

Reply:
I mean

```suggestion
data            = flags payload-length payload padding [signature] [iv] [tag]
```

Reply:
That is confusing because it doesn't represent the encrypted part. It also doesn't capture a similar thing happening to asymmetric encryption... please re-read this and related parts.

Reply:
The alternative would be to pull out encrypted field and use that, in which case it'd be something like

`encrypted = cipher iv tag`

I didn't do this because

1) It is standard for these primitives and mentioned
2) It'd require doing it for ECIES too, and seems like it'd easily get messy/wrong; I'd rather defer such details to other specs, and every implementer should use an existing library for this. We aren't trying to reinvent the wheel here.

Reply:
Sorry, I realize I am trying to second guess things instead of just pointing out what I don't understand:
- what is `data` here?
- what is `auxiliary-field?`

Reply:
Oh right, auxiliary-field should indeed be payload-length, forgot to change that. `data` is what is being encrypted, that can probably be clarified above. Thanks!

Reply:
```suggestion
data            = flags payload-length payload padding [signature]
```